diesel/src/serialize.rs

```diff
@@ -30,43 +30,34 @@ pub enum IsNull {
 }
 
 /// Wraps a buffer to be written by `ToSql` with additional backend specific
-/// utilities.
-#[derive(Clone, Copy)]
+/// utilities.y
 pub struct Output<'a, T, DB>
 where
     DB: TypeMetadata,
     DB::MetadataLookup: 'a,
 {
     out: T,
-    metadata_lookup: Option<&'a DB::MetadataLookup>,
+    metadata_lookup: Option<&'a mut DB::MetadataLookup>,
 }
 
 impl<'a, T, DB: TypeMetadata> Output<'a, T, DB> {
     /// Construct a new `Output`
-    pub fn new(out: T, metadata_lookup: &'a DB::MetadataLookup) -> Self {
+    pub fn new(out: T, metadata_lookup: &'a mut DB::MetadataLookup) -> Self {
         Output {
             out,
             metadata_lookup: Some(metadata_lookup),
         }
     }
 
-    /// Create a new `Output` with the given buffer
-    pub fn with_buffer<U>(&self, new_out: U) -> Output<'a, U, DB> {
-        Output {
-            out: new_out,
-            metadata_lookup: self.metadata_lookup,
-        }
-    }
-
     /// Return the raw buffer this type is wrapping
     pub fn into_inner(self) -> T {
         self.out
     }
 
     /// Returns the backend's mechanism for dynamically looking up type
     /// metadata at runtime, if relevant for the given backend.
-    pub fn metadata_lookup(&self) -> &'a DB::MetadataLookup {
-        self.metadata_lookup.expect("Lookup is there")
+    pub fn metadata_lookup(&mut self) -> &mut &'a mut DB::MetadataLookup {
+        self.metadata_lookup.as_mut().expect("Lookup is there")
```

Comment:
`&mut &mut` is not necessary using reborrow:
```suggestion
    pub fn metadata_lookup(&mut self) -> &mut DB::MetadataLookup {
        &mut **self.metadata_lookup.as_mut().expect("Lookup is there")
```
but with the Option seemingly not being necessary either:
```rust
    pub fn metadata_lookup(&mut self) -> &mut DB::MetadataLookup {
        &mut *self.metadata_lookup
    }
```